src/Core/Framework/Adapter/Cache/CacheTagCollector.php

```diff
@@ -39,23 +44,38 @@ public function reset(): void
     }
 
     /**
-     * @return array<string>
+     * Collects cache tags for the current request, which will be used to tag the http cache entry.
+     * This method will prevent adding the same tag multiple times and will not dispatch an event if only existing tags are provided.
      */
-    public function get(Request $request): array
+    public function addTag(string ...$tags): void
```

Comment:
would you mind adding the new method under the get method? so git diff is better readable :see_no_evil: